pennylane/gradients/general_shift_rules.py

```diff
@@ -378,6 +381,44 @@ def generate_multi_shift_rule(frequencies, shifts=None, orders=None):
     return _combine_shift_rules(rules)
 
 
+def _copy_and_shift_params(tape, indices, shifts, multipliers, cast=False):
+    """Create a copy of a tape and of parameters, and set the new tape to the parameters
+    rescaled and shifted as indicated by ``indices``, ``multipliers`` and ``shifts``."""
+    all_ops = tape.circuit
+
+    for idx, shift, multiplier in zip(indices, shifts, multipliers):
+        _, op_idx, p_idx = tape.get_operation(idx)
+        op = (
+            all_ops[op_idx].obs
+            if isinstance(all_ops[op_idx], MeasurementProcess)
+            else all_ops[op_idx]
+        )

```

Comment:
why not this?
```suggestion
        op, op_idx, p_idx = tape.get_operation(idx)
```

Reply:
I don't understand why, but making the suggested change causes the gradient to be incorrect ❓ ❓

Reply:
Incorrect how? When I made a similar change a while back, it actually fixed a bug where the gradients were previously incorrect.

Reply:
we need to get it from `all_ops` in case the same operation has multiple params, since `all_ops` would have been changed in a previous loop iteration. lgtm!